### DISCUSSION TITLE
Stream stops periodically and resumes

### DISCUSSION THREAD

**Discussion by DannyDarth** [streamlink/streamlink-twitch-gui#930] [General]
I know you've gotten many threads on this. I see that streamlink itself just updated and i have updated to 5.1.0. 
But streams still stop during commercial breaks and then resumes. Just wanted to know if this is intended when mid-roll ads
are present, or is it suppose to just keep playing with no stream interruption?

**Comment by bastimeyer** [comment]
Read this
https://github.com/streamlink/streamlink-twitch-gui/issues/754
https://github.com/streamlink/streamlink/issues/4949

**Reply by DannyDarth** [reply]
So its a no on preventing them :/

**Comment by KaXaSA** [comment]
Definitely not as good as using:
`Streamlink Twitch Gui + Streamlink + MVP/Other video player`

**BUT**, if you want to avoid having to wait on that purple screen, you could try: 
`Streamlink Twitch Gui + Microsoft Edge + Tampermonkey (Browser Extension) + Purple Adblocker (User Script)`

\- if it stops working, then you will have to look for something to replace Purple Adblocker

\- It should work with chrome, but the parameters mentioned below could be different

\- uBlock Origin and maybe other AD blockers could cause issues like **black screen loop, stream never loading...**
So if you have issues, try to disable the AD Blockers, other browser Extensions/Add-ons and keep only the Purple Adblocker (User Script)

---
After installing Tampermonkey and adding the user script:
Open [ Streamlink Twitch Gui > Settings > Player ]


1. **Player Preset**:
```
No Preset
```

2.  **Video Player** (Browser Location):
```
msedge.exe
OR
C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe
```

3. **Parameters**:

[Windowed] Opens Edge in a new window without url/toolbar, WITH title bar. (I couldn't find a way to launch it without the title bar):
```
--app=http://www.twitch.tv/"{channel}"
```

**OR**

[Fullscreen] Open Edge FullScreen ([Kiosk Mode](https://learn.microsoft.com/en-us/deployedge/microsoft-edge-configure-kiosk-mode))

```
--kiosk http://www.twitch.tv/"{channel}" --edge-kiosk-type=fullscreen
```

(I didn't test kiosk mode, maybe there is a better way to launch it fullscreen, I'm not sure)

---

Good to know:

- **Theater Mode** - BTTV (extension/add-on) has an option to automatically open streams in Theater Mode.
- **Windows Position** - Microsoft Powertoys > FancyZones allow you to open most of the windows in a specific position/size, useful if you want to launch something like the Twitch Stream (without chat) + Chatty, side by side.
- **Stretch Video** - Remove black bars from the stream video, but depending on the dimensions/aspect ratio it can look a little distorted. Use _Stylus_ or some other browser extension/addon that allows you to customize/inject CSS (most Chrome extension also works on Edge). Create a new userstyle for _twitch.tv_ with this inside:
```
.video-player video { object-fit: fill !important; }
```
